sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -162,6 +162,15 @@ class PlannerSuite extends SparkFunSuite with SQLTestUtils {
     assert(planned.head.isInstanceOf[execution.TakeOrderedAndProject])
   }
 
+  test("Remove repartition operators when they are the child of Exchange and shuffle=True") {
```

Comment:
For completeness, we also need a test case to ensure that this pruning does _not_ apply if `shuffle=False`.